packages/edit-post/src/components/layout/index.js

```diff
@@ -225,7 +225,7 @@ function Layout( { styles } ) {
 							<TextEditor />
 						) }
 						{ isRichEditingEnabled && mode === 'visual' && (
-							<VisualEditor styles={ styles } />
+							<VisualEditorOrPluginPreview styles={ styles } />
```

Comment:
It feels like `PluginPreview` should be independent of the `mode`. Well, maybe at least it should still be possible to see the preview when `isRichEditingEnabled` is disabled. It needs some more thinking because depending on the content of the preview, it might not play with how we use the visual editor today.

Could you disable rich editing, use text mode to edit content and still use the custom preview from the plugin?